### PR TITLE
Test cases for Rest Client

### PR DESCRIPTION
**Pull request by rajdeepd**

### CODE REVIEW

**Comment by rajdeepd** [comment]
@deads2k kindly review

**Comment by deads2k** [comment]
@rajdeepd Sorry, vacation.  Doesn't seem like there's a way to explain that to the bot :)

Can you add some description?  I'm not against more tests, but I'd like to understand what has led you here.  The existing tests were written based on knowledge of the error handling code which is consistent for all types so only one needed to be tested.

File: pkg/client/restclient/client_test.go [hunk 1]
```diff
@@ -156,6 +158,203 @@ func TestDoRequestCreated(t *testing.T) {
...
+	status := &unversioned.Status{Status: unversioned.StatusSuccess}
+	expectedBody, _ := runtime.Encode(testapi.Default.Codec(), status)
+	fakeHandler := utiltesting.FakeHandler{
+		StatusCode:   202,
+		ResponseBody: string(expectedBody),
+		T:            t,
+	}
+	testServer := httptest.NewServer(&fakeHandler)
+	defer testServer.Close()
+	c, err := RESTClientFor(&Config{
```
Comment (deads2k) [inline pkg/client/restclient/client_test.go]: This stanza is repeated a lot in the existing test and in yours, helper method?

**Comment by rajdeepd** [comment]
I am trying to test Http Methods not tested with the current set of tests

File: pkg/client/restclient/client_test.go [hunk 2]
```diff
@@ -156,6 +158,203 @@ func TestDoRequestCreated(t *testing.T) {
...
+	if err == nil {
+		t.Errorf("Expected Error")
+	}
+	if statusOut != nil {
+		t.Errorf("Unexpected mis-match. Expected %#v.  Saw %#v", nil, statusOut)
+	}
+	fakeHandler.ValidateRequest(t, "/"+testapi.Default.GroupVersion().String()+"/test", "GET", nil)
+}
+
+func TestHttpMethods(t *testing.T) {
```
Comment (deads2k) [inline pkg/client/restclient/client_test.go]: This is unnecessary.  If these are ever nil the controller and client tests will just fall over

File: pkg/client/restclient/client_test.go [hunk 3]
```diff
@@ -156,6 +158,203 @@ func TestDoRequestCreated(t *testing.T) {
...
+		t.Errorf("Delete : Object returned should not be nil")
+	}
+
+	request = c.Patch(api.JSONPatchType)
+	if(request == nil){
+		t.Errorf("Patch : Object returned should not be nil")
+	}
+}
+
+func TestConfigGroupVersionNil(t *testing.T) {
```
Comment (deads2k) [inline pkg/client/restclient/client_test.go]: which group version in this test is nil?

File: pkg/client/restclient/client_test.go [hunk 4]
```diff
@@ -156,6 +158,203 @@ func TestDoRequestCreated(t *testing.T) {
 	fakeHandler.ValidateRequest(t, "/"+testapi.Default.GroupVersion().String()+"/test", "GET", nil)
 }
 
+func TestDoRequestNotCreated(t *testing.T) {
+	status := &unversioned.Status{Status: unversioned.StatusSuccess}
```
Comment (deads2k) [inline pkg/client/restclient/client_test.go]: expectedStatus

File: pkg/client/restclient/client_test.go [hunk 5]
```diff
@@ -156,6 +158,203 @@ func TestDoRequestCreated(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("Unexpected error %#v", err)
+	}
+	if !reflect.DeepEqual(status, statusOut) {
+		t.Errorf("Unexpected mis-match. Expected %#v.  Saw %#v", status, statusOut)
+	}
+	fakeHandler.ValidateRequest(t, "/"+testapi.Default.GroupVersion().String()+"/test", "GET", nil)
+}
+
+func TestDoRequestAcceptedNoContentReturned(t *testing.T) {
```
Comment (deads2k) [inline pkg/client/restclient/client_test.go]: Collapse all these `WasCreated` tests down into the `testCases := []struct{}` pattern and I'll be ok having multiples.  I'm still not certain they're testing different code paths, but in a sufficiently compact representation it's fine.

**Comment by rajdeepd** [comment]
@deads2k fixed as suggested ..

**Comment by rajdeepd** [comment]
@deads2k do i need to do something here?

**Comment by deads2k** [comment]
> @deads2k do i need to do something here?

No, the queue is only open for 1.3 items.  When it opens back up this should progress.

**Pull request merged**